src/callbacks_step/amr_dg2d.jl

```diff
@@ -1,3 +1,95 @@
+# Redistribute data for load balancing after partitioning the mesh
+function redistribute_data!(u_ode::AbstractVector, mesh::TreeMesh{2}, equations, 
+                            dg::DGSEM, cache, old_mpi_ranks_per_cell)
+  # Retain current solution data
+  old_n_elements = nelements(dg, cache)
+  old_cell_ids = copy(cache.elements.cell_ids)
+  old_u_ode = copy(u_ode)
+  GC.@preserve old_u_ode begin # OBS! If we don't GC.@preserve old_u_ode, it might be GC'ed
+    old_u = wrap_array(old_u_ode, mesh, equations, dg, cache)
+
+    reinitialize_containers!(mesh, equations, dg, cache)
+    @unpack elements = cache
+
+    resize!(u_ode, nvariables(equations) * nnodes(dg)^ndims(mesh) * nelements(dg, cache))
+    u = wrap_array(u_ode, mesh, equations, dg, cache)
+
+    # Get new list of leaf cells
+    leaf_cell_ids = local_leaf_cells(mesh.tree)
+
+    # Collect MPI requests for MPI_Waitall
```

Comment:
Wrap the code section starting from this line up to and including the `MPI.Waitall!` command in a `begin ... end` block and time it with something like `@timeit_debug timer() "exchange data" begin...`
This way we will know how much overhead is created by the exchange routine alone.

Reply:
Good idea!